woodwork/table_accessor.py

```diff
@@ -1122,6 +1126,111 @@ def pearson_correlation(
         )
         return pd.DataFrame(pearson_dict)
 
+    @_check_table_schema
+    def spearman_correlation_dict(
```

Comment:
Is there any way to consolidate these at all to reduce total lines? could we just take in a correlation_type as an input?

Reply:
@Cmancuso According to @rwedge, we previously had `mutual_info` and `mutual_info_dict` as accessor methods before `dependence` and `dependence_dict` were added to WW. When he added Pearsons, he extended these accessor methods for continuity, and I decided to extend Spearman also for continuity. These are all supported purely by `dependence` and `dependence_dict` methods, but consolidating them could require a breaking change.